connectors/sources/mongo.py

```diff
@@ -106,6 +112,18 @@ def __init__(self, configuration):
             client_params["username"] = user
             client_params["password"] = password
 
+        if self.configuration["ssl_enabled"]:
+            client_params["tls"] = True
+            if ssl_ca:
+                pem_certificates = get_pem_format(key=ssl_ca)
+                with NamedTemporaryFile(mode="w", suffix=".pem", delete=False) as cert:
+                    cert.write(pem_certificates)
+                    self.certfile = cert.name
```

Comment:
Can we also organise cleanup of certificates?

This is gonna spam file system with certificates every time connector class is created - in reality if you have 1 connector configured, you'll have 1 temporary file every 30 seconds -> 720 files per day -> 262800 files per year per connector. That's too much.

Reply:
Can we pass `delete_on_close=True` to `NamedTemporaryFile`? I wonder, is passing a file the only way? We already have the CA content, can we just pass it to `AsyncIOMotorClient`?

Reply:
Is the `close` method reliably called everywhere that a DataSource instance is instantiated? I think that's Artem's concern - that every heartbeat/ping/rule-validation instance is not calling `close()`

Reply:
@seanstory we need to double check, last time I was checking, call of `validate` on connector never called `close` after validation is ended.

Additionally, all connectors with SSL that I've checked are affected with this resource leakage.

Reply:
Hi @parthpuri-elastic , could you investigate whether the temp file can be safely deleted after the `AsyncIOMotorClient` is instantiated?